distributed/shuffle/_arrow.py

```diff
@@ -47,21 +47,40 @@ def check_minimal_arrow_version() -> None:
 
 
 def convert_shards(shards: list[pa.Table], meta: pd.DataFrame) -> pd.DataFrame:
+    import pandas as pd
     import pyarrow as pa
+    from pandas.core.dtypes.cast import find_common_type  # type: ignore[attr-defined]
 
     from dask.dataframe.dispatch import from_pyarrow_table_dispatch
 
-    table = pa.concat_tables(shards)
+    table = pa.concat_tables(shards, promote_options="permissive")
 
     df = from_pyarrow_table_dispatch(meta, table, self_destruct=True)
-    return df.astype(meta.dtypes, copy=False)
+    reconciled_dtypes = {}
+    for column, dtype in meta.dtypes.items():
+        actual = df[column].dtype
+        if actual == dtype:
+            continue
```

Comment:
Interestingly, the dtypes of the `_partitions` column don't match up (`int64` vs `uint64`). @phofl, can pandas handle this conversion as a zero-copy operation?